File: Flask/microservices/route1-service/website/distance/distance_dto.py
from static.utils import attributes


class GetDistanceDtoRequest:
    def __init__(self, placeId1: int, placeId2: int, coordinates1: str, coordinates2: str) -> None:
        self.placeId1 = placeId1
        self.placeId2 = placeId2
        self.coordinates1 = coordinates1
        self.coordinates2 = coordinates2
    
    def as_dict(self) -> dict:
        return attributes(self)

File: Spring Boot/microservices/route2-service/src/main/java/pl/edu/pg/benchmarking/route2/distance/DistanceDto.java
package pl.edu.pg.benchmarking.route2.distance;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

public class DistanceDto {

    @AllArgsConstructor(access = AccessLevel.PUBLIC)
    @Getter
    public static class GetDistanceRequest {
        private int placeId1;
        private int placeId2;
        private String coordinates1;
        private String coordinates2;
    }

    @Getter
    public static class GetDistanceResponse {
        private int placeId1;
        private int placeId2;
        private Double distance;
    }

}
